crates/api_common/src/request.rs

```diff
@@ -97,11 +97,12 @@ pub fn generate_post_link_metadata(
   context: Data<LemmyContext>,
 ) {
   spawn_try_task(async move {
+    // Decide if the thumbnail should be generated
     let allow_sensitive = local_site_opt_to_sensitive(&local_site);
     let page_is_sensitive = post.nsfw;
     let allow_generate_thumbnail = allow_sensitive || !page_is_sensitive;
-    let mut thumbnail_url = custom_thumbnail.or_else(|| post.thumbnail_url.map(Into::into));
-    let do_generate_thumbnail = thumbnail_url.is_none() && allow_generate_thumbnail;
+    let do_generate_thumbnail =
+      allow_generate_thumbnail && custom_thumbnail.is_none() && post.thumbnail_url.is_none();
```

Comment:
> Specifically if you set a custom thumbnail when creating a post, then edit the post body it will overwrite with auto-generated thumbnail.

I fixed this check here to prevent that, and tested it. It will not fetch a new metadata pic if the post.thumbnail_url already exists.